tests/release/e2e-test-default-with-deployment.go

```diff
@@ -203,6 +203,18 @@ var _ = framework.ReleaseSuiteDescribe("[HACBS-1199]test-release-e2e-with-deploy
 			}, releasePipelineRunCompletionTimeout, defaultInterval).Should(BeTrue())
 		})
 
+		// phase Deployed happens before phase Released
+		It("tests a Release should report the deployment was successful.", func() {
```

Comment:
In general, it looks good, thanks for improving this. But I have a comment.

It seems both this test case (`It(...`) and the next one are almost the same - the only difference is that one checks `IsDeployed()` and the other `IsReleased()`. I was wondering if we could split this into three test cases:

1. Check that we can get the Release
2. Wait for it to be Deployed
3. Wait for it to be Released

But we would still need to call `GetFirstReleaseInNamespace()` in 2. and 3., right? So probably we can keep it as is. But maybe at least let's unite the two tests descriptions? So I would suggest this for the two tests:

1. `It("tests a Release exists and was deployed successfuly."`
2. `It("tests a Release exists and was released successfuly."`

Or something like that.

Or maybe we could actually combine these into a single test case, since, based on the linked slack chat, both statuses should happen at the same time. But that's just an idea. Maybe it's clearer if there are two test cases. It's up to you.

Reply:
Thank you so much for the suggestion, @mmalina!

I follow the suggestion of splitting it into three test cases to be easier to track, in case something goes wrong.